pydrive2/test/test_fs.py

```diff
@@ -25,17 +26,39 @@ def remote_dir(base_remote_dir):
     return base_remote_dir + "/" + str(uuid.uuid4())
 
 
-@pytest.fixture
-def fs(tmpdir, base_remote_dir):
+@pytest.fixture(scope="module")
+def service_auth(tmp_path_factory):
     setup_credentials()
-    auth = GoogleAuth(settings_file_path("default.yaml", tmpdir / ""))
+    tmpdir = tmp_path_factory.mktemp("settings")
+    auth = GoogleAuth(settings_file_path("default.yaml", wkdir=tmpdir))
     auth.ServiceAuth()
+    return auth
+
+
+@pytest.fixture(scope="module")
+def fs_factory(base_remote_dir, service_auth):
```

Comment:
hmm, what is the alternative? I can still explore that in the followup ...

Reply:
You can initialize the `base_remote_dir` automatically in another fixture (with `module` scope and use it in `fs`. This will only be called once. 

```python
@pytest.fixture(scope="module")
def base_dir(remote_dir):
    fs = GDriveFileSystem(base_remote_dir, service_auth)
    fs._gdrive_create_dir(...)
    yield remote_dir
    # cleanup

@pytest.fixture
def fs(base_dir):
   ...
```

Then, you can just do `fs = GDriveFileSystem(base_remote_dir, service_auth)` wherever you are doing `fs_factory()`, right? Or, did I miss something here?

Reply:
okay, I see. There are a few other things (e.g. I'm disabling fs caching in the fixture), probably it all can be moved to the base_dir or something, but I'm not sure it make a big difference 

it's easier tbh for me to have all this logic in one place - easier to do changes.